### DISCUSSION TITLE
Caveats with the `anon_key`, or am I misunderstanding something?

### DISCUSSION THREAD

**Discussion by rwacarter** [supabase/supabase#21320] [Questions]
I'm trying to get my head around the fact that the `anon_key` can be exposed in the browser (via `NEXT_PUBLIC_ANON_KEY` or similar), and not be a huge security issue, even with RLS. 

For example, let's say I have a function that lets a user write a comment on another user's article. Maybe the function does a load of quite complex checks, maybe involving some third party services or something, to see if the user can write the comment, and then, if everything's OK, adds the comment to the `comments` table. 

With the publicly available `anon_key` however, can't anyone just use that to add any comment they want? Do the complex checks in the original code have to be duplicated in their entirety as RLS policies, which in some cases may not even be possible? Or have I missed something fundamental in how the `anon_key` works, and exposing it in the browser has no security implications?

**Comment by GaryAustin1** [comment]
You normally would have signed in users and do RLS on that to protect your tables.  If you allow anon access with RLS then anyone can access your info or update it if your RLS allows it.

**Reply by rwacarter** [reply]
Thanks for your reply. So I'm right in thinking that any limitations you want to put on users accessing or manipulating data, no matter how complex or involved, has to be done using RLS policies?

**Reply by GaryAustin1** [reply]
That or you need to use edge functions or server side code and have a simple lock in RLS.  Then your protected code can do security as you see fit.
You can use functions and table joins in RLS so it can cover a lot of cases.
